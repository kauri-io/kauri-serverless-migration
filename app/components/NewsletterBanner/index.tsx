import TextField from '@material-ui/core/TextField'
import Button from '../../components/Button'
import * as Yup from 'yup'
import { useState, FunctionComponent } from 'react'
import { Grid, makeStyles, Theme, Typography } from '@material-ui/core'

const useStyles = makeStyles((theme: Theme) => {
    return {
        container: {
            display: 'flex',
            backgroundColor: 'white',
        },
        title: {
            fontSize: 20,
            fontWeight: 500,
            lineHeight: '24px',
        },
        subtitle: {
            fontSize: 14,
            fontWeight: 400,
            maxWidth: 400,
        },
        button: {},
        input: {
            maxWidth: 400,
            width: '100%',
        },
        newsletterSignup: {
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'flex-start',
            background: `url(${global.window &&
                global.window.location
                    .origin}/static/images/NewsletterSVG.svg) no-repeat`,
            width: 1224,
            margin: 'auto',
            alignItems: 'flex-start',
            padding: theme.spacing(2),
        },
    }
})

interface IProps {
    handleSubmit: (emailAddress: string) => void
    handleError: () => void
}

interface IState {
    emailAddress: string | null
}

const handleEmailAddress = (
    handleError: () => void,
    handleSubmit: (emailAddress: string) => void,
    state: IState
) => async () => {
    const validate = Yup.string()
        .email()
        .required()
    const result = await validate.isValid(state.emailAddress)
    if (result && state.emailAddress) {
        return handleSubmit(state.emailAddress)
    } else {
        handleError()
    }
}

const NewsletterBanner: FunctionComponent<IProps> = ({
    handleError,
    handleSubmit,
}) => {
    const [state, setState] = useState<IState>({
        emailAddress: '',
    })

    const classes = useStyles()

    return (
        <div className={classes.container}>
            <Grid container={true} className={classes.newsletterSignup}>
                <Typography variant="h4" className={classes.title}>
                    Kauri Newsletter
                </Typography>
                <Typography variant="body1" className={classes.subtitle}>
                    Subscribe below and receive the latest Ethereum tutorials
                    and project announcements every 2 weeks!
                </Typography>
                <TextField
                    fullWidth={true}
                    margin="normal"
                    placeholder={'Enter your email address'}
                    onChange={({ target: { value: emailAddress } }) =>
                        setState({ emailAddress })
                    }
                    onKeyPress={(e: React.KeyboardEvent) => {
                        if (e.key === 'Enter') {
                            handleEmailAddress(
                                handleError,
                                handleSubmit,
                                state
                            )().catch(_ => {
                                return
                            })
                        }
                    }}
                    InputProps={{
                        className: classes.input,
                    }}
                    value={state.emailAddress}
                />
                <Button
                    color="primary"
                    onClick={handleEmailAddress(
                        handleError,
                        handleSubmit,
                        state
                    )}
                    className={classes.button}
                >
                    Subscribe
                </Button>
            </Grid>
        </div>
    )
}

export default NewsletterBanner
