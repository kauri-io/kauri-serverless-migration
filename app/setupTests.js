import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
