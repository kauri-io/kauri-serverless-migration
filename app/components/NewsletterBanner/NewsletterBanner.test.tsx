import NewsletterBanner from './index'
import { mountWithRedux } from '../../setupTests'

describe('components/NewsletterBanner', () => {
    it('should match snapshot', () => {
        const NewsletterBannerProps = {
            handleSubmit: jest.fn(),
            handleError: jest.fn(),
        }

        const wrapper = mountWithRedux(
            <NewsletterBanner {...NewsletterBannerProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
