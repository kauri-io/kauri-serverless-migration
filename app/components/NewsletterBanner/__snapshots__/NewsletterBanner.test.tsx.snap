// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NewsletterBanner should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <NewsletterBanner
      handleError={[MockFunction]}
      handleSubmit={[MockFunction]}
    >
      <NewsletterBanner__Container>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "NewsletterBanner__Container-sc-1kyqtrr-0",
                "isStatic": false,
                "lastClassName": "iWaUWO",
                "rules": Array [
                  "display:flex;width:100%;justify-content:center;align-items:center;padding:",
                  [Function],
                  "px;background:",
                  [Function],
                  ";",
                ],
              },
              "displayName": "NewsletterBanner__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "NewsletterBanner__Container-sc-1kyqtrr-0",
              "target": "section",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <section
            className="NewsletterBanner__Container-sc-1kyqtrr-0 iWaUWO"
          >
            <NewsletterBanner__Content>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "NewsletterBanner__Content-sc-1kyqtrr-1",
                      "isStatic": false,
                      "lastClassName": "isoakR",
                      "rules": Array [
                        "display:flex;width:500px;flex-direction:column;justify-content:center;align-items:center;>:not(:last-child){margin-bottom:",
                        [Function],
                        "px;}>:nth-child(2){margin-bottom:",
                        [Function],
                        "px;}",
                      ],
                    },
                    "displayName": "NewsletterBanner__Content",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "NewsletterBanner__Content-sc-1kyqtrr-1",
                    "target": "section",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <section
                  className="NewsletterBanner__Content-sc-1kyqtrr-1 isoakR"
                >
                  <Icon>
                    <svg
                      fill="none"
                      height="52"
                      viewBox="0 0 64 52"
                      width="64"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 2H56C59.3 2 62 4.7 62 8V44C62 47.3 59.3 50 56 50H8C4.7 50 2 47.3 2 44V8C2 4.7 4.7 2 8 2Z"
                        stroke="#0BA986"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="4"
                      />
                      <path
                        d="M62 8L32 29L2 8"
                        stroke="#0BA986"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="4"
                      />
                    </svg>
                  </Icon>
                  <Typography__styledComponent
                    color="white"
                  >
                    <StyledComponent
                      color="white"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Typography__styledComponent-sc-36xkf6-5",
                            "isStatic": false,
                            "lastClassName": "jNVtUZ",
                            "rules": Array [
                              "color:",
                              [Function],
                              ";margin:0px;font-weight:",
                              "500",
                              ";",
                              ";font-size:",
                              "20",
                              "px;text-transform:",
                              [Function],
                              ";text-align:",
                              [Function],
                              ";:hover{color:",
                              [Function],
                              ";cursor:",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "Typography__styledComponent",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="Typography__styledComponent-sc-36xkf6-5 jNVtUZ"
                        color="white"
                      >
                        Kauri Newsletter
                      </span>
                    </StyledComponent>
                  </Typography__styledComponent>
                  <Typography__component
                    color="white"
                    textAlign="center"
                  >
                    <StyledComponent
                      color="white"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Typography__component-sc-36xkf6-4",
                            "isStatic": false,
                            "lastClassName": "chprKh",
                            "rules": Array [
                              "font-size:14px;font-weight:normal;letter-spacing:-0.1px;margin-bottom:",
                              [Function],
                              "px;line-height:18px;color:",
                              [Function],
                              ";",
                              ";",
                              [Function],
                              ";",
                            ],
                          },
                          "displayName": "Typography__component",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Typography__component-sc-36xkf6-4",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      textAlign="center"
                    >
                      <span
                        className="Typography__component-sc-36xkf6-4 chprKh"
                        color="white"
                      >
                        Subscribe below and receive the latest Ethereum tutorials and project announcements every 2 weeks!
                      </span>
                    </StyledComponent>
                  </Typography__component>
                  <Input
                    color="white"
                    fontSize={3}
                    handleChange={[Function]}
                    onKeyPress={[Function]}
                    placeHolder="Enter your email address"
                    textAlign="center"
                  >
                    <Input__Wrapper
                      textAlign="center"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "Input__Wrapper-sc-100s1kb-2",
                              "isStatic": false,
                              "lastClassName": "cAYXgm",
                              "rules": Array [
                                "display:flex;position:relative;align-self:auto;",
                                [Function],
                                " ",
                                [Function],
                                " ",
                                [Function],
                              ],
                            },
                            "displayName": "Input__Wrapper",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "Input__Wrapper-sc-100s1kb-2",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        textAlign="center"
                      >
                        <div
                          className="Input__Wrapper-sc-100s1kb-2 cAYXgm"
                        >
                          <Input__InputComp
                            color="white"
                            fontSize={3}
                            fontWeight={400}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            onKeyPress={[Function]}
                            onKeyUp={[Function]}
                            placeholder="Enter your email address"
                            textAlign="center"
                            value=""
                          >
                            <StyledComponent
                              color="white"
                              fontSize={3}
                              fontWeight={400}
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Input__InputComp-sc-100s1kb-0",
                                    "isStatic": false,
                                    "lastClassName": "kGODxV",
                                    "rules": Array [
                                      "text-align:",
                                      [Function],
                                      ";font-size:",
                                      [Function],
                                      "px;font-weight:",
                                      [Function],
                                      ";color:",
                                      [Function],
                                      ";background:transparent;width:100%;outline:none;border:none;::-webkit-input-placeholder{color:",
                                      [Function],
                                      ";}:focus::-webkit-input-placeholder{text-indent:-999px;}",
                                    ],
                                  },
                                  "displayName": "Input__InputComp",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Input__InputComp-sc-100s1kb-0",
                                  "target": "input",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyPress={[Function]}
                              onKeyUp={[Function]}
                              placeholder="Enter your email address"
                              textAlign="center"
                              value=""
                            >
                              <input
                                className="Input__InputComp-sc-100s1kb-0 kGODxV"
                                color="white"
                                fontSize={3}
                                fontWeight={400}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyPress={[Function]}
                                onKeyUp={[Function]}
                                placeholder="Enter your email address"
                                value=""
                              />
                            </StyledComponent>
                          </Input__InputComp>
                          <Input__Underline
                            fontSize={3}
                            textAlign="center"
                          >
                            <StyledComponent
                              fontSize={3}
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Input__Underline-sc-100s1kb-1",
                                    "isStatic": false,
                                    "lastClassName": "hyDdeC",
                                    "rules": Array [
                                      "user-select:none;border-top:2px solid ",
                                      [Function],
                                      ";position:",
                                      [Function],
                                      " left:0;bottom:0;max-width:100%;height:0px;color:transparent;overflow:hidden;font-size:",
                                      [Function],
                                      "px;margin-left:2px;",
                                    ],
                                  },
                                  "displayName": "Input__Underline",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Input__Underline-sc-100s1kb-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              textAlign="center"
                            >
                              <span
                                className="Input__Underline-sc-100s1kb-1 hyDdeC"
                                fontSize={3}
                              >
                                Enter your email address
                              </span>
                            </StyledComponent>
                          </Input__Underline>
                        </div>
                      </StyledComponent>
                    </Input__Wrapper>
                  </Input>
                  <PrimaryButtonComponent
                    onClick={[Function]}
                  >
                    <PrimaryButton
                      bg="primary"
                      color="white"
                      fontSize={0}
                      fontWeight={700}
                      mr={2}
                      onClick={[Function]}
                      type="submit"
                    >
                      <StyledComponent
                        bg="primary"
                        color="white"
                        fontSize={0}
                        fontWeight={700}
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "PrimaryButton-sc-1cbxa53-0",
                              "isStatic": false,
                              "lastClassName": "bVbdII",
                              "rules": Array [
                                "display:flex;align-items:center;justify-content:center;min-width:136px;max-width:150px;",
                                [Function],
                                ";height:40px;border-radius:4px;border:none;cursor:pointer;> svg,img{height:16px;",
                                [Function],
                                ";}text-transform:uppercase;opacity:",
                                [Function],
                                ";",
                                [Function],
                                ";",
                                [Function],
                                ";",
                                [Function],
                                ";",
                                [Function],
                                ";:hover{cursor:",
                                [Function],
                                ";}",
                                ";:hover{background-color:",
                                [Function],
                                ";}",
                              ],
                            },
                            "displayName": "PrimaryButton",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "PrimaryButton-sc-1cbxa53-0",
                            "target": "button",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        mr={2}
                        onClick={[Function]}
                        type="submit"
                      >
                        <button
                          className="PrimaryButton-sc-1cbxa53-0 bVbdII"
                          color="white"
                          fontSize={0}
                          fontWeight={700}
                          onClick={[Function]}
                          type="submit"
                        >
                          Subscribe
                        </button>
                      </StyledComponent>
                    </PrimaryButton>
                  </PrimaryButtonComponent>
                </section>
              </StyledComponent>
            </NewsletterBanner__Content>
          </section>
        </StyledComponent>
      </NewsletterBanner__Container>
    </NewsletterBanner>
  </ThemeProvider>
</Provider>
`;
